### PR TITLE
Minor fixes and tweaks to the enclave/evm

### PR DESCRIPTION
**Pull request by otherview**
### Why is this change needed?

- part of https://github.com/obscuronet/obscuro-internal/issues/596
- Fixes minor bugs that would be reported when using the deployment tools for uniswap
-
### What changes were made as part of this PR:

- evm_facade.go 
   - Error log upgrade
   - ExecuteOffChainCall now uses a pointer to address to support offchain contract creation calls
   - ExecuteOffChainCall executes messages with a default 100M gas

- rollup_chain.go
   - Correctly handles nil strings
- rpc_utils.go
   - ExtractCallParamTo correctly extracts nil addresses ( contract creation ) 

### :rotating_light: Definition of Done :rotating_light:
- [ ] Deployed into dev-testnet 
- [ ] Passes tests on dev-testnet
- [ ] [Changelog.md](https://github.com/obscuronet/go-obscuro/blob/main/docs/testnet/changelog.md) updated

### CODE REVIEW

File: go/enclave/rpc/rpc_utils.go [hunk 1]
```diff
@@ -60,20 +60,26 @@ func GetViewingKeyAddressForTransaction(tx *common.L2Tx) (gethcommon.Address, er
...
-func ExtractCallParamTo(callParams []byte) (gethcommon.Address, error) {
+func ExtractCallParamTo(callParams []byte) (*gethcommon.Address, error) {
 	var paramsJSONMap []interface{}
 	err := json.Unmarshal(callParams, &paramsJSONMap)
 	if err != nil {
-		return gethcommon.Address{}, fmt.Errorf("could not parse JSON params in eth_call request. JSON params are: %s. Cause: %w", string(callParams), err)
+		return nil, fmt.Errorf("could not parse JSON params in eth_call request. JSON params are: %s. Cause: %w", string(callParams), err)
 	}
 
-	txArgs := paramsJSONMap[0] // The first argument is the transaction arguments, the second the block, the third the state overrides.
```
Comment (joeldudley) [inline go/enclave/rpc/rpc_utils.go:70]: Can you leave this comment in?

Comment (otherview) [reply go/enclave/rpc/rpc_utils.go:70]: Good point. Moved it to where it was previously, makes more sense.

File: go/enclave/evm/evm_facade.go [hunk 2]
```diff
@@ -55,12 +61,12 @@ func executeTransaction(s *state.StateDB, cc *params.ChainConfig, chain *Obscuro
...
 
 // ExecuteOffChainCall - executes the "data" command against the "to" smart contract
-func ExecuteOffChainCall(from gethcommon.Address, to gethcommon.Address, data []byte, s *state.StateDB, header *common.Header, rollupResolver db.RollupResolver, chainConfig *params.ChainConfig) (*gethcore.ExecutionResult, error) {
+func ExecuteOffChainCall(from gethcommon.Address, to *gethcommon.Address, data []byte, s *state.StateDB, header *common.Header, rollupResolver db.RollupResolver, chainConfig *params.ChainConfig) (*gethcore.ExecutionResult, error) {
 	chain, vmCfg, gp := initParams(rollupResolver, true)
 
 	blockContext := gethcore.NewEVMBlockContext(convertToEthHeader(header), chain, &header.Agg)
 	// todo use ToMessage
-	msg := types.NewMessage(from, &to, 0, gethcommon.Big0, 100_000, gethcommon.Big0, gethcommon.Big0, gethcommon.Big0, data, nil, true)
+	msg := types.NewMessage(from, to, 0, gethcommon.Big0, 100_000_000_000, gethcommon.Big0, gethcommon.Big0, gethcommon.Big0, data, nil, true)
```
Comment (joeldudley) [inline go/enclave/evm/evm_facade.go:69]: Can you add a comment explaining what `100_000_000_000` represents?

**Review by joeldudley — APPROVED** [review]
Just some small changes.

**Pull request merged**